### PR TITLE
Api client part of ci build

### PR DESCRIPTION
**Pull request by MartinBentley**
### Context
This is the story to ensure the nuget package is generated as part of the ci process.
Here's the card for this:https://trello.com/c/B6mKdaOm/533-build-searchandcompare-apiclient-as-part-of-ci
### Changes proposed in this pull request
The changes in this PR also include a change to the project name from SearchAndCompareDomain to SearchAndCompareApiClient.
### Guidance to review
 In addition the namespaces should also need to be changed from Domain to ApiClient however this caused many issues and I ended up reverting the namespace change. This still needs to be done but with another card after this change goes in.

apiclientaspartofbuild

### CODE REVIEW

File: SearchAndCompareApi.sln [hunk 1]
```diff
@@ -19,12 +17,16 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution
 		Dockerfile = Dockerfile
 		Dockerfile.rhel7 = Dockerfile.rhel7
 		global.json = global.json
+		nuget.config = nuget.config
```
Comment (defong) [inline SearchAndCompareApi.sln]: Is there a need?

File: SearchAndCompareApi.sln [hunk 2]
```diff
@@ -19,12 +17,16 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution
 		Dockerfile = Dockerfile
 		Dockerfile.rhel7 = Dockerfile.rhel7
 		global.json = global.json
+		nuget.config = nuget.config
 		package.json = package.json
 		README.md = README.md
 		travis-build.sh = travis-build.sh
 		travis-deploy.sh = travis-deploy.sh
+		travis-nuget-deploy.sh = travis-nuget-deploy.sh
```
Comment (defong) [inline SearchAndCompareApi.sln]: can we change the filename to reflect its only api client only

File: src/api/SearchAndCompareApi.csproj [hunk 3]
```diff
@@ -25,7 +25,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\domain\SearchAndCompareDomain.csproj" />
+    <ProjectReference Include="..\domain\SearchAndCompareApiClient.csproj" />
```
Comment (defong) [inline src/api/SearchAndCompareApi.csproj]: do folder directory need to change?

Comment (MartinBentley) [reply src/api/SearchAndCompareApi.csproj]: That's in a new card along with namespace change

File: SearchAndCompareApi.sln (hunk 2 above)
Comment (MartinBentley) [reply SearchAndCompareApi.sln]: No!

Comment (MartinBentley) [reply SearchAndCompareApi.sln]: It's consistent with the identical file in manage courses

File: SearchAndCompareApi.sln (hunk 1 above)
Comment (MartinBentley) [reply SearchAndCompareApi.sln]: Yes it is needed

**Pull request merged**